nav2_route/include/nav2_route/corner_smoothing.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2025, Polymath Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_ROUTE__CORNER_SMOOTHING_HPP_
+#define NAV2_ROUTE__CORNER_SMOOTHING_HPP_
+
+#include <vector>
+#include <cmath>
+
+#include "nav2_route/types.hpp"
+#include "nav2_route/utils.hpp"
+
+namespace nav2_route
+{
+class CornerArc
```

Comment:
In this class: convert all doubles to floats, there's mixing going on which has burned me in the past in planners. I don't think anything here needs double percision, so you could just search + replace for this file and be OK